### PR TITLE
Quick sort

### PR DESCRIPTION
**Pull request by cahudson94**
Pull for submission.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -277,6 +277,15 @@ All modules and tests located in src directory.
...
+        - if more then one type of input is given return error message.
+
+
+## Quick Sort
+- module: quick_sort.py
+- test: test_quick_sort.py
+- public methods:
+    ### quicksort:
         - sort the input list.
         - if more then one type of input is given return error message.
```
Comment (clair3st) [inline README.md]: Please include Big O time complexity

File: src/working/quick_sort.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            di += 1
+
+            while len(right) > ri:
+                data[di] = right[ri]
+                ri += 1
+                di += 1
+
+        return data
+
+    except TypeError:
```
Comment (clair3st) [inline src/working/quick_sort.py]: Again, weird to return just a string and not raise an error.

File: src/working/quick_sort.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    try:
+        if len(data) > 1:
+            mid = data[0]
+            left = []
+            right = []
+
+            for item in data[1:]:
+                # This slows it down by almost a second but covers 2.7
+                if not isinstance(item, type(mid)):
+                    raise TypeError
```
Comment (clair3st) [inline src/working/quick_sort.py]: If you raise a TypeError, use `raise TypeError('some text')` to help with error handling

File: src/working/test_quick_sort.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+from random import randint
+
+
+TEST_PARAMS = [
+    ([i for i in range(50)]),
+    ([randint(1, 500) for i in range(50)]),
+    ([5, 9, 1, 2, 6]),
+    ([234, 23, 52, 66]),
+    ([4, 3, 2, 1]),
+    ([99, 22, 55, 4, 66, 87, 23, 11]),
```
Comment (clair3st) [inline src/working/test_quick_sort.py]: Add tests with negative numbers and duplicates.
